src/snowflake/snowpark/modin/plugin/__init__.py

```diff
@@ -89,3 +90,13 @@
 snowflake.snowpark._internal.utils.should_warn_dynamic_pivot_is_in_private_preview = (
     False
 )
+
+
+# TODO: SNOW-1504302: Modin upgrade - use Snowpark pandas DataFrame for isocalendar
+def isocalendar(self):  # type: ignore
+    from snowflake.snowpark.modin.pandas import DataFrame
+
+    return DataFrame(query_compiler=self._query_compiler.dt_isocalendar())
```

Comment:
can you add a comment explaining the exact reason for adding this method? you can copy/paste this

```
OSS Modin's DatetimeProperties frontend class wraps the returned query compiler with `modin.pandas.DataFrame`. Since we currently replace `pd.DataFrame` with our own Snowpark pandas DataFrame object, this causes errors since OSS Modin explicitly imports its own DataFrame class here.
This override can be removed once the frontend DataFrame class is removed from our codebase.
```

Reply:
Good idea, thanks for the writeup -- done!